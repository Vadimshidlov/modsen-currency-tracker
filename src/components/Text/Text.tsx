import React, { Component } from "react";
import { TextPropsType } from "@/types/TimeLinePageTypes";

export default class Text extends Component<TextPropsType> {
    render() {
        const { children, className, ...rest } = this.props;

        return (
            <p className={className} {...rest}>
                {children}
            </p>
        );
    }
}
