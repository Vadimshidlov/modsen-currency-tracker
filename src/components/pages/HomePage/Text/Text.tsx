import React from "react";
import { TextPropsType } from "@/types/HomePageTypes/types";

export default function Text({ className, children, ...rest }: TextPropsType) {
    return (
        <p className={className} {...rest}>
            {children}
        </p>
    );
}
