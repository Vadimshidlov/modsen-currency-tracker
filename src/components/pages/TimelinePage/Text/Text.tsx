import React, { Component } from "react";
import { TextPropsType } from "@/types/TimeLinePageTypes/types";

export default class Text extends Component<TextPropsType> {
    render() {
        const { children, className, ...rest } = this.props;

        return (
            <p className={className} {...rest}>
                {children}
            </p>
        );
    }
}
