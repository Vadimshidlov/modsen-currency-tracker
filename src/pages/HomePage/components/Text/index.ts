export { default } from "@/pages/HomePage/components/Text/Text";
