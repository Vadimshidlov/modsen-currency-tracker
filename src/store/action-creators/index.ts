export { getMapData } from "@/store/action-creators/getMapData";
export { getCurrency } from "@/store/action-creators/getCurrency";
export {
    selectSecondCurrency,
    closeModalWindow,
    openModalWindow,
} from "@/store/action-creators/openModalWindow";
export { switchLightTheme, switchDarkTheme } from "@/store/action-creators/toggleTheme";
export {
    errorAuthUserLocation,
    successAuthUserLocation,
} from "@/store/action-creators/authUserLocation";
