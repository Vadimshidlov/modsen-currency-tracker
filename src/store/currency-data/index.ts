export { mockMapData } from "@/store/currency-data/mockMapData";
export { mockBanksApiData } from "@/store/currency-data/mockBanksApiData";
export { mockStockData } from "@/store/currency-data/mockStockData";
