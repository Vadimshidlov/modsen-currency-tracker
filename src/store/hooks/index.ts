export { useActions } from "@/store/hooks/useActions";
export { useTypedSelectorHook } from "@/store/hooks/useTypedSelector";
