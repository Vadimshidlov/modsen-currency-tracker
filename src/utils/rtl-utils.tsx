import React, { PropsWithChildren } from "react";
import { Store, applyMiddleware, createStore } from "redux";
import { thunk } from "redux-thunk";
import { render } from "@testing-library/react";
import type { RenderOptions } from "@testing-library/react";
import { Provider } from "react-redux";
import { RootStateType, rootReducer } from "@/store/reducers";

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
    preloadedState?: Partial<RootStateType>;
    store?: Store;
}

export function renderWithProviders(
    ui: React.ReactElement,
    {
        preloadedState = {},
        store = createStore(rootReducer, preloadedState, applyMiddleware(thunk)),
        ...renderOptions
    }: ExtendedRenderOptions = {},
) {
    function Wrapper({ children }: PropsWithChildren<object>): JSX.Element {
        return <Provider store={store}>{children}</Provider>;
    }
    return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from "@testing-library/react";

export { renderWithProviders as render };
