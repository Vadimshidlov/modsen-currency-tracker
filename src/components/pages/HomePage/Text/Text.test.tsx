import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "@/store";
import { renderWithProviders, screen } from "@/utils/rtl-utils";
import Text from "@/components/pages/HomePage/Text/Text";

describe("CurrencyStocks component tests", () => {
    it("should correct render component", async () => {
        renderWithProviders(<Text className="text-class">Hello!</Text>);

        expect(screen.getByText("Hello!")).toBeInTheDocument();
        expect(screen.getByText("Hello!")).toHaveClass("text-class");
    });

    it("snapshot matches", async () => {
        const { asFragment } = renderWithProviders(<Text className="text-class">Hello!</Text>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("should match snapshot", () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <Text className="text-class">Hello!</Text>
                </Provider>,
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
