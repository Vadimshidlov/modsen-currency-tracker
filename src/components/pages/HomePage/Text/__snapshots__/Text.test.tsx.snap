// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyStocks component tests should match snapshot 1`] = `
<p
  className="text-class"
>
  Hello!
</p>
`;

exports[`CurrencyStocks component tests snapshot matches 1`] = `
<DocumentFragment>
  <p
    class="text-class"
  >
    Hello!
  </p>
</DocumentFragment>
`;
